src/transactions/CategoryFilter.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { PropTypes } from 'prop-types';
+import { CATEGORIES } from '../data';
+
+function CategoryFilter({ handleFilter }) {
+  function onSelectChanged(event) {
+    const { value } = event.target;
+
+    handleFilter({ categoryId: value });
+  }
+
+  function onClear() {
+    handleFilter({ categoryId: null });
+  }
+
+  return (
+    <div className="row">
+      <div className="col col-3 mr-2">
+        <select className="form-select" name="categoryFilter" onChange={onSelectChanged}>
+          <option>Filter by Category</option>
+          {CATEGORIES.map((category) => (
+            <option key={category.id} value={category.id} name={category.name}>
+              {category.name}
+            </option>
+          ))}
+        </select>
+      </div>
+      <div className="col col-3">
+        <button className="btn btn-outline-secondary" type="submit" onClick={onClear}>Clear Filter</button>
```

Comment:
Right now when the user clicks the Clear Filter button, the select still shows the previously selected filter. That could be confusing to the user. Can we make it so the select goes back to the Filter by Category option when they clear the filter? This is a bit of a trickier change, but I think it would be a real UX improvement. Especially because we're planning to use this component in multiple places.

You'd probably need to pass the categoryId part of the filter state down to this component from TransactionFilters. It looks like you're already passing the filter state to TransactionFilters, you're just not using it.